docker/lib/dependabot/docker/version.rb

```diff
@@ -13,54 +14,65 @@ module Docker
     # for a description of Java versions.
     #
     class Version < Dependabot::Version
+      extend T::Sig
       # The regex has limits for the 0,255 and 1,255 repetitions to avoid infinite limits which makes codeql angry.
       # A docker image cannot be longer than 255 characters anyways.
       DOCKER_VERSION_REGEX = /^(?<prefix>[a-z._\-]{0,255})[_\-v]?(?<version>.{1,255})$/
 
+      sig { override.params(version: T.nilable(T.any(String, Integer, Gem::Version))).void }
```

Comment:
I had completely forgotten that we had defined a [type alias][1] for this big type. It's kinda long, and this makes it easier: 

```suggestion
      sig { override.params(version: VersionParameter).void }
```

[1]: https://sorbet.org/docs/type-aliases